sakura_core/CGrepEnumFileBase.h

```diff
@@ -115,8 +115,9 @@ class CGrepEnumFileBase {
 	int Enumerates( LPCWSTR lpBaseFolder, VGrepEnumKeys& vecKeys, CGrepEnumOptions& option, CGrepEnumFileBase* pExceptItems = NULL ){
 		int found = 0;
 
+		size_t szBaseFolder = wcslen(lpBaseFolder);
```

Comment:
`CGrepEnumFileBase::Enumerates` is a public function, then `lpBaseFolder` can be NULL. 

It is desirable to take measures against nullptr as follows.
```suggestion
		const auto cchBaseFolder = lpBaseFolder ? wcslen(lpBaseFolder) : 0;
```

It is more preferable to take measures against missing the NUL termination as follows.
```suggestion
		const auto cchBaseFolder = lpBaseFolder ? wcsnlen_s(lpBaseFolder, _MAX_PATH - 1) : 0;
```
Sakura-Editor can not handle paths above _MAX_PATH.

Reply:
@berryzplus fix it https://github.com/sakura-editor/sakura/pull/1860/commits/1fc1801a340c0d1ebb2b306d27b3d48c14ec2c17